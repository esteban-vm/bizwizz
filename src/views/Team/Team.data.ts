import type { TeamMateProps } from './TeamMate'

const teamMates: TeamMateProps[] = [
  { id: 'fec6d982-1994-4d02-9cc7-18a86719fb7c', name: 'Paula Lee', designation: 'CEO' },
  { id: 'c512f1ff-6fff-4aee-be97-7c0697204186', name: 'Ted Johnson', designation: 'Co-Founder' },
  { id: '37d5c8a6-3234-42f1-8379-65b50f5f12cb', name: 'Miranda Shaffer', designation: 'Lead Designer' },
]

export default teamMates
