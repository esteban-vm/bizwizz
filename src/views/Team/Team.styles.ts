import clsx from 'clsx'

const styles = {
  wrapper: clsx('flex flex-wrap justify-evenly'),
}

export default styles
