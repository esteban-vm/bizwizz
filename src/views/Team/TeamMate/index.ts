export { default, type TeamMateProps } from './TeamMate'
