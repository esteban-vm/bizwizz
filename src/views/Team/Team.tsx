import type { FC } from '@/types'
import { Wave } from '@/components'
import TeamMate, { type TeamMateProps } from './TeamMate'
import styles from './Team.styles'

export const teamMates: TeamMateProps[] = [
  { id: 'fec6d982-1994-4d02-9cc7-18a86719fb7c', name: 'Paula Lee', designation: 'CEO' },
  { id: 'c512f1ff-6fff-4aee-be97-7c0697204186', name: 'Ted Johnson', designation: 'Co-Founder' },
  { id: '37d5c8a6-3234-42f1-8379-65b50f5f12cb', name: 'Miranda Shaffer', designation: 'Lead Designer' },
]

const Team: FC = () => {
  return (
    <section aria-labelledby='team_title' className='section bg-black' id='team'>
      <Wave is='white' />
      <div className='section_header'>
        <h2 className='section_heading' id='team_title'>
          Meet our team
        </h2>
        <h3 className='section_subheading'>Lorem ipsum dolor sit amet</h3>
      </div>
      <div className={styles.team_mates}>
        {teamMates.map((teamMate) => (
          <TeamMate key={teamMate.id} {...teamMate} />
        ))}
      </div>
    </section>
  )
}

export default Team
