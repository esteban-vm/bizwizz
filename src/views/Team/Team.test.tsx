import { render, cleanup, within, screen } from '@/tests'
import Team, { teamMates } from './Team'

describe('Team:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Team />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /meet our team/i
    const subtitle = /lorem ipsum dolor sit amet/i
    const teamElem = screen.getByRole('region')
    expect(teamElem).toBeInTheDocument()
    expect(teamElem).toHaveAccessibleName(title)
    expect(teamElem).toHaveAttribute('id', 'team')
    expect(within(teamElem).getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(teamElem).getByRole('heading', { name: title, level: 2 })).toBeInTheDocument()
    expect(within(teamElem).getByRole('heading', { name: subtitle, level: 3 })).toBeInTheDocument()
    expect(within(teamElem).getAllByRole('article')).toHaveLength(teamMates.length)
  })
})
