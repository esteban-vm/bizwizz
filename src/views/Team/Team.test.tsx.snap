// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Team: > should render correctly 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="team_title"
    class="section bg-black"
    id="team"
  >
    <svg
      aria-hidden="true"
      class="h-40 w-full text-white"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <div
      class="section_header"
    >
      <h2
        class="section_heading"
        id="team_title"
      >
        Meet our team
      </h2>
      <h3
        class="section_subheading"
      >
        Lorem ipsum dolor sit amet
      </h3>
    </div>
    <div
      class="flex flex-wrap justify-evenly"
    >
      <article
        class="relative h-96 w-96 select-none rounded-full border-[3px] border-solid border-primary bg-cover bg-center first:bg-[url('/images/team-mate-1.jpg')] last:bg-[url('/images/team-mate-3.jpg')] even:bg-[url('/images/team-mate-2.jpg')] [&>div:last-child]:hover:top-1/2 [&>div]:hover:opacity-100"
        id="fec6d982-1994-4d02-9cc7-18a86719fb7c"
      >
        <div
          class="absolute left-0 top-0 h-full w-full rounded-full bg-primary/70 opacity-0 transition-opacity duration-200"
        />
        <div
          class="absolute left-1/2 top-[30%] -translate-x-1/2 -translate-y-1/2 text-white opacity-0 transition-all duration-500"
        >
          <h4
            class="text-[2.7rem] font-semibold"
          >
            Paula Lee
          </h4>
          <p
            class="text-2xl font-medium"
          >
            CEO
          </p>
          <div
            class="mt-8 flex justify-between"
          >
            <a
              href="https://www.facebook.com"
              target="_blank"
              title="facebook"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-facebook"
                />
              </svg>
            </a>
            <a
              href="https://www.twitter.com"
              target="_blank"
              title="twitter"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-twitter"
                />
              </svg>
            </a>
            <a
              href="https://www.linkedin.com"
              target="_blank"
              title="linkedin"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-linkedin"
                />
              </svg>
            </a>
          </div>
        </div>
      </article>
      <article
        class="relative h-96 w-96 select-none rounded-full border-[3px] border-solid border-primary bg-cover bg-center first:bg-[url('/images/team-mate-1.jpg')] last:bg-[url('/images/team-mate-3.jpg')] even:bg-[url('/images/team-mate-2.jpg')] [&>div:last-child]:hover:top-1/2 [&>div]:hover:opacity-100"
        id="c512f1ff-6fff-4aee-be97-7c0697204186"
      >
        <div
          class="absolute left-0 top-0 h-full w-full rounded-full bg-primary/70 opacity-0 transition-opacity duration-200"
        />
        <div
          class="absolute left-1/2 top-[30%] -translate-x-1/2 -translate-y-1/2 text-white opacity-0 transition-all duration-500"
        >
          <h4
            class="text-[2.7rem] font-semibold"
          >
            Ted Johnson
          </h4>
          <p
            class="text-2xl font-medium"
          >
            Co-Founder
          </p>
          <div
            class="mt-8 flex justify-between"
          >
            <a
              href="https://www.facebook.com"
              target="_blank"
              title="facebook"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-facebook"
                />
              </svg>
            </a>
            <a
              href="https://www.twitter.com"
              target="_blank"
              title="twitter"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-twitter"
                />
              </svg>
            </a>
            <a
              href="https://www.linkedin.com"
              target="_blank"
              title="linkedin"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-linkedin"
                />
              </svg>
            </a>
          </div>
        </div>
      </article>
      <article
        class="relative h-96 w-96 select-none rounded-full border-[3px] border-solid border-primary bg-cover bg-center first:bg-[url('/images/team-mate-1.jpg')] last:bg-[url('/images/team-mate-3.jpg')] even:bg-[url('/images/team-mate-2.jpg')] [&>div:last-child]:hover:top-1/2 [&>div]:hover:opacity-100"
        id="37d5c8a6-3234-42f1-8379-65b50f5f12cb"
      >
        <div
          class="absolute left-0 top-0 h-full w-full rounded-full bg-primary/70 opacity-0 transition-opacity duration-200"
        />
        <div
          class="absolute left-1/2 top-[30%] -translate-x-1/2 -translate-y-1/2 text-white opacity-0 transition-all duration-500"
        >
          <h4
            class="text-[2.7rem] font-semibold"
          >
            Miranda Shaffer
          </h4>
          <p
            class="text-2xl font-medium"
          >
            Lead Designer
          </p>
          <div
            class="mt-8 flex justify-between"
          >
            <a
              href="https://www.facebook.com"
              target="_blank"
              title="facebook"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-facebook"
                />
              </svg>
            </a>
            <a
              href="https://www.twitter.com"
              target="_blank"
              title="twitter"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-twitter"
                />
              </svg>
            </a>
            <a
              href="https://www.linkedin.com"
              target="_blank"
              title="linkedin"
            >
              <svg
                class="m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light"
                role="img"
              >
                <use
                  xlink:href="#icon-linkedin"
                />
              </svg>
            </a>
          </div>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
